pkg/asset/userprovided.go

```diff
@@ -0,0 +1,39 @@
+package asset
+
+import (
+	"fmt"
+)
+
+// UserProvided generates an asset that is supplied by a user.
+type UserProvided struct {
+	Prompt string
+}
+
+var _ Asset = (*UserProvided)(nil)
+
+// Dependencies returns no dependencies.
+func (a *UserProvided) Dependencies() []Asset {
+	return []Asset{}
+}
+
+// Generate queries for input from the user.
+func (a *UserProvided) Generate(map[Asset]*State) (*State, error) {
+	input := queryUser(a.Prompt)
+	return &State{
+		Contents: []Content{
+			{Data: []byte(input)},
+		},
+	}, nil
+}
+
+func queryUser(prompt string) string {
+	for {
+		fmt.Print(prompt)
+		var input string
+		if _, err := fmt.Scanln(&input); err != nil {
```

Comment:
It seems like [`Scanln`][1] is breaking on whitespace (regardless of newline-ness).  I think we may need to use a [`Scanner.Scan`][2] (although there are other options as well).  [Here's a play][3] showing the issue (note the "expected newline" error on the first `Fscanln`).

For most of our inputs (email addresses, domain names, ...), we don't want spaces in the input.  But the pull secret is JSON, which may contain spaces.  And folks could have spaces in their password(/phrase).  So where we want to enforce "this line contains no spaces" (or "this line contains an `@` and other valid email-address structure"), I'd rather leave that to explicit validation, instead of having it be a limit imposed by the generic `queryUser`.

[1]: https://golang.org/pkg/fmt/#Scanln
[2]: https://golang.org/pkg/bufio/#example_Scanner_lines
[3]: https://play.golang.org/p/IZVgFdvjzz7